File: solutions/0001-two-sum/solution.py
# 1. Two Sum
# https://leetcode.com/problems/two-sum/
# Accepted: 2026-09-18T14:41:53.000Z
# Language: Python3
# Runtime: 3 ms · Beats 53.49%
# Memory: 20.9 MB · Beats 7.4%
# Submission: https://leetcode.com/submissions/detail/2145830840/

class Solution:
    def twoSum(self, nums: list[int], target: int) -> list[int]:
        
        seen = {}
        for i, num in enumerate(nums) : 
            if num in seen : 
                return [seen[num], i]
            seen[target - num] = i

File: solutions/0001-two-sum/solution.cpp
// 1. Two Sum
// https://leetcode.com/problems/two-sum/
// Accepted: 2026-09-24T20:32:18.000Z
// Language: C++
// Runtime: 3 ms · Beats 67.86%
// Memory: 15.4 MB · Beats 7.73%
// Submission: https://leetcode.com/submissions/detail/2152424213/

class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int, int> seen ; 
        seen.reserve(nums.size()) ; 
        for (int i=0 ; i< static_cast<int>(nums.size()) ; i++) {
            if (auto it = seen.find(target - nums[i]); it!= seen.end()){
                return {it->second, i} ;
            }
            seen.emplace(nums[i],i);
        }
        return {} ;
    }
};
